packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -935,6 +921,32 @@ abstract class DatabaseInstanceSource extends DatabaseInstanceNew implements IDa
       target: this,
     });
   }
+
+  private addSingleUserRotationForMasterSecret(options: RotationSingleUserOptions): secretsmanager.SecretRotation {
+    if (!this.secret) {
+      throw new Error('Cannot add single user rotation for an instance without master secret.');
+    }
+
+    return this.addSingleUserRotationForSecret(this, 'RotationSingleUser', this.secret, options);
+  }
+
+  private addSingleUserRotationForSecret(scope: Construct, id: string, secret: secretsmanager.ISecret, options: RotationSingleUserOptions)
+    : secretsmanager.SecretRotation {
+    if (Node.of(scope).tryFindChild(id)) {
+      const secretName = this === scope ? 'master' : 'additional'; // Is there a better way to do this?
+      throw new Error(`A single user rotation for ${secretName} secret was already added to this instance.`);
```

Comment:
Same comment here (include more details about the Secret that's causing the failure).